### DISCUSSION TITLE
How to setup vue component testing for cy.mount with JSX?

### DISCUSSION THREAD

**Discussion by dante01yoon** [cypress-io/cypress#28401] [Questions and Help]
Greetings, I want to write component test using jsx syntax without  pass all the props with objects stated in official docs. How to write component test like below?
<img width="481" alt="스크린샷 2023-11-24 오후 9 39 41" src="https://github.com/cypress-io/cypress/assets/6510430/41afe528-3c90-47ca-a957-76ff7ebe12a2">

**Comment by portyu9** [comment]
Hi @dante01yoon 

Yes — current Cypress Vue Component Testing supports this JSX form directly:

```jsx
cy.mount(
  <Stepper
    initial={100}
    onChange={onChangeSpy}
  />
)
```

instead of:

```js
cy.mount(Stepper, {
  props: {
    initial: 100,
    onChange: onChangeSpy,
  },
})
```

The important requirement is that your **Component Testing bundler must support Vue JSX/TSX**.

For example, with Vite:

```js
// vite.config.js
import { defineConfig } from 'vite'
import vue from '@vitejs/plugin-vue'
import vueJsx from '@vitejs/plugin-vue-jsx'

export default defineConfig({
  plugins: [
    vue(),
    vueJsx(),
  ],
})
```

Install the plugin if needed:

```bash
npm install -D @vitejs/plugin-vue-jsx
```

Then your spec can be a JSX/TSX file:

```jsx
import Stepper from './Stepper.vue'

describe('Stepper', () => {
  it('mounts using JSX', () => {
    cy.mount(<Stepper initial={100} />)

    cy.contains('100').should('be.visible')
  })
})
```

Cypress's Vue mount adapter accepts either the Vue component/options form or a JSX element. If JSX does not compile, the problem is normally the Vite/Webpack JSX transformation configuration rather than `cy.mount()` itself.

Please mark this answer as accepted if it helped, thank you.
